From 48b6681419f939232f2ed1bf2f90d1f9fc2d2b62 Mon Sep 17 00:00:00 2001
From: Robin Lee <rgl@google.com>
Date: Fri, 10 Sep 2021 12:03:42 +0000
Subject: [PATCH] Initialize DrawingState::trustedOverlay to false in
 constructor

To avoid it being initialised to true randomly.

Bug: 199483370
Bug: 196389741
Bug: 218605327
Change-Id: I75be2b1d305e22f8a71532b9f5b8ea6c469baaaa
Merged-In: I75be2b1d305e22f8a71532b9f5b8ea6c469baaaa
(cherry picked from commit edb375df82f6b48688d85c11dbca91adef088bc8)
(cherry picked from commit b4a2d3a77ab2fc8b7e3d88c528c71269651f0ddb)
(cherry picked from commit c058b4851d187d10a8f134c02c5cf1a22c1a9d39)
Merged-In: I75be2b1d305e22f8a71532b9f5b8ea6c469baaaa
---
 services/surfaceflinger/Layer.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/surfaceflinger/Layer.cpp b/services/surfaceflinger/Layer.cpp
index 4d804fa67..3e50217b8 100644
--- a/services/surfaceflinger/Layer.cpp
+++ b/services/surfaceflinger/Layer.cpp
@@ -135,6 +135,7 @@ Layer::Layer(const LayerCreationArgs& args)
     mDrawingState.postTime = -1;
     mDrawingState.destinationFrame.makeInvalid();
     mDrawingState.dropInputMode = gui::DropInputMode::NONE;
+    mDrawingState.isTrustedOverlay = false;
 
     if (args.flags & ISurfaceComposerClient::eNoColorFill) {
         // Set an invalid color so there is no color fill.
-- 
2.40.0.rc1.284.g88254d51c5-goog

